api/db/changelog/db.changelog-5-authority.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="mwf" id="changelog-5-authority">
+    <createTable tableName="authority">
```

Comment:
You're missing a primary key on this table? It should be user_id, authority.

Reply:
MySQL automatically gives you an index on user id (it's part of the primary key, and also is a foreign key -- double whammy indexing.)

You won't get an index on authority unless you add one. But this table is going to be tiny, so a scan is not a big deal; and it may be that any e-mail we send out (assuming that is the right workflow) could just go to some alias maintained externally... I don't know that we need to read all of these.